mos-platform/osi-c1p/getchar.c

```diff
@@ -1,7 +1,18 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-int getchar() {
-  puts("getchar() not yet implemented");
-  abort();
+extern unsigned char volatile __CHARBUF;
+
+int getchar(void)
+{
+    do
+    {
+        asm volatile ("JSR\t__kbhit" ::: "p", "a", "x", "y");
```

Comment:
If I understand you correctly writing this as a C call this would clobber imaginary registers, and this would generate additional save and restore code for these registers. If that is the case, the `asm` call does save some code.

Reply:
Only if the compiler actually does decide to put anything in the imaginary registers for this; should be easy to check in the generated assembly.